core/embed/projects/prodtest/cmd/secure_channel.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "secure_channel.h"
+
+#include "string.h"
+
+typedef enum {
+  SECURE_CHANNEL_STATE_0,  // Handshake has not been initiated yet
+  SECURE_CHANNEL_STATE_1,  // Handshake in progress (after calling
+                           // `secure_channel_handshake_1()` and before calling
+                           // `secure_channel_handshake_2()`)
+  SECURE_CHANNEL_STATE_2   // Handshake completed (after calling
+                           // `secure_channel_handshake_2()`),
+                           // `secure_channel_encrypt()` can be called
+} noise_state_t;
```

Comment:
Wouldn't it actually make more sense for this `enum` to replace the `bool initialized` in `noise_context_t`?